game/src/plugins/skills/fishing/src/fishing.plugin.kts

```diff
@@ -0,0 +1,133 @@
+import org.apollo.cache.def.ItemDefinition
+import org.apollo.game.action.DistancedAction
+import org.apollo.game.message.impl.NpcActionMessage
+import org.apollo.game.model.Position
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.Skill
+import org.apollo.game.plugin.skills.fishing.*
+import kotlin.properties.Delegates
+import java.util.Random
+
+class FishingAction: DistancedAction<Player> {
+
+    var spot: FishingSpot by Delegates.notNull()
+    var tool: FishingTool by Delegates.notNull()
+    var started: Boolean = false
+    var options: IntArray by Delegates.notNull()
+    var minLevel: Int = 110
+
+    constructor(mob: Player, position: Position, spot: FishingSpot, option: Int) : super(4, true, mob, position, 1) {
+        this.tool = FISHING_TOOLS[spot.tools[option - 1]]!!
+        this.spot = spot
+        this.options = if (option == 1) spot.firstFish else spot.secondFish
```

Comment:
I don't think this should be done in the constructor.  Maybe pass it in instead.